src/component/ProfileEditForm.tsx

```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
 
 function ProfileEditForm({ onClose, onSave }: ProfileEditFormProps){
     const { register, formState: {errors}, handleSubmit } = useForm<ProfileEditFormInput>();
+    const {user} = useUserStore();
+    const updateUser = useUserStore(state => state.updateUser);
 
-    const onSubmit: SubmitHandler<ProfileEditFormInput> = (data) => {
-        console.log(data);
-        onSave(data);
-    }
+    const onSubmit: SubmitHandler<ProfileEditFormInput> = async (data, e) => {
+        e?.preventDefault();
+        const userData: User = {
+            id: 0,
```

Comment:
Still don't really like using 0 for id. Try to just use `user.id` that you have from store